### PR TITLE
Feature/add api audit listener

### PR DESCRIPTION
**Pull request by alexnetflix**
Custom AOP implementation could be used to audit all REST and gRPC using this annotation.

### CODE REVIEW

File: grpc-server/src/main/java/com/netflix/conductor/grpc/server/service/HealthServiceImpl.java [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 import io.grpc.health.v1.HealthCheckRequest;
 import io.grpc.health.v1.HealthCheckResponse;
 import io.grpc.health.v1.HealthGrpc;
 import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
 
+@Audit
```
Comment (s3nthil) [inline grpc-server/src/main/java/com/netflix/conductor/grpc/server/service/HealthServiceImpl.java]: Audit is typically for logging user access to resources. Is it necessary to audit health check?

File: grpc-server/src/main/java/com/netflix/conductor/grpc/server/service/TaskServiceImpl.java [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
 import org.slf4j.LoggerFactory;
 
 import javax.inject.Inject;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
+@Audit
```
Comment (s3nthil) [inline grpc-server/src/main/java/com/netflix/conductor/grpc/server/service/TaskServiceImpl.java]: I think this has methods used by client for polling and other system operations. Should those be audited?

File: jersey/src/main/java/com/netflix/conductor/server/resources/HealthCheckResource.java [hunk 3]
```diff
@@ -10,12 +12,11 @@
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 
-import io.swagger.annotations.Api;
-
 @Api(value = "/health", produces = MediaType.APPLICATION_JSON, tags = "Health Check")
 @Path("/health")
 @Produces({MediaType.APPLICATION_JSON})
 @Singleton
+@Audit
```
Comment (s3nthil) [inline jersey/src/main/java/com/netflix/conductor/server/resources/HealthCheckResource.java]: Same comment as others - is this necessary?

File: grpc-server/src/main/java/com/netflix/conductor/grpc/server/service/HealthServiceImpl.java (hunk 1 above)
Comment (alexnetflix) [reply grpc-server/src/main/java/com/netflix/conductor/grpc/server/service/HealthServiceImpl.java]: hey sorry, I accidentally shared this PR with you, I removed all the reviewers yesterday but somehow missed you :) this PR is not quite ready yet...

**Pull request closed without merging**